src/stdlib_strings.f90

```diff
@@ -644,38 +654,48 @@ end function padl_char_char
     !> Right pad the input string with the 'pad_with' string
     !>
     !> Returns a new string
-    pure function padr_string_string(string, output_length, pad_with) result(res)
+    pure function padr_string_default(string, output_length) result(res)
         type(string_type), intent(in) :: string
         integer, intent(in) :: output_length
-        type(string_type), intent(in) :: pad_with
         type(string_type) :: res
 
-        res = string_type(padr_char_char(char(string), output_length, char(pad_with)))
-    end function padr_string_string
+        res = string_type(padr_char_char(char(string), output_length, " "))
+    end function padr_string_default
 
     !> Right pad the input string with the 'pad_with' string
     !>
     !> Returns a new string
-    pure function padr_string_char(string, output_length, pad_with) result(res)
+    pure function padr_string_pad_with(string, output_length, pad_with) result(res)
         type(string_type), intent(in) :: string
         integer, intent(in) :: output_length
         character(len=1), intent(in) :: pad_with
         type(string_type) :: res
 
         res = string_type(padr_char_char(char(string), output_length, pad_with))
-    end function padr_string_char
+    end function padr_string_pad_with
 
     !> Right pad the input string with the 'pad_with' string
     !>
     !> Returns a new string
-    pure function padr_char_string(string, output_length, pad_with) result(res)
+    pure function padr_char_default(string, output_length) result(res)
         character(len=*), intent(in) :: string
         integer, intent(in) :: output_length
-        type(string_type), intent(in) :: pad_with
         character(len=max(len(string), output_length)) :: res
 
-        res = padr_char_char(string, output_length, char(pad_with))
-    end function padr_char_string
+        res = padr_char_char(string, output_length, " ")
```

Comment:
For the default it should be possible to do simply
```
res = string
```
getting rid of one function and branch (if) call.

But I would need to consult the Fortran standard to guarantee the padding symbol will be whitespace under all possible circumstances.

Reply:
on page 162 of https://j3-fortran.org/doc/year/18/18-007r1.pdf I have got this:
![image](https://user-images.githubusercontent.com/34789087/123508974-64d13900-d690-11eb-883f-4c316addcbd3.png)

Reply:
The second bullet point sounds like what we want. Thanks for finding and referencing this.